File: app/src/main/java/com/rachelleignacio/listbucket/domain/interactors/AddListItemInteractor.java
package com.rachelleignacio.listbucket.domain.interactors;

import com.rachelleignacio.listbucket.domain.interactors.base.Interactor;
import com.rachelleignacio.listbucket.domain.models.ListItem;

/**
 * Created by rachelleignacio on 3/9/17.
 */

public interface AddListItemInteractor extends Interactor {
    interface  Callback {
        void onListItemAdded(ListItem newItem);
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/domain/interactors/AddListItemInteractor.kt
package com.rachelleignacio.listbucket.domain.interactors

import com.rachelleignacio.listbucket.domain.interactors.base.Interactor
import com.rachelleignacio.listbucket.domain.models.ListItem

/**
 * Created by rachelleignacio on 8/9/17.
 */
interface AddListItemInteractor : Interactor {
    interface Callback {
        fun onListItemAdded(newItem: ListItem)
    }
}